packages/eslint-plugin/src/rules/no-unnecessary-type-parameters.ts

```diff
@@ -210,6 +210,7 @@ function collectTypeParameterUsageCounts(
   function visitType(
     type: ts.Type | undefined,
     assumeMultipleUses: boolean,
+    isReturnType = false,
```

Comment:
Does `isTypeParameterRepeatedInAST` need a corresponding update?

Reply:
Don't think so. With the proposed change, single uses of Array and ReadonlyArray always defer the check at a later step, the one with type information. If the single use is as parameter we get an error, whereas the single use is as return type it's fine.

Reply:
I added some tests [here](https://github.com/typescript-eslint/typescript-eslint/pull/9536/commits/fe7efb154c010dbb348fe336ae5b9e0fd8192045#diff-bb675ade954bc4c24d9bf6172222cad53e9d22776514ebf4ea8189015b57b3c3R214-R218) to check everything is fine for `Array` and `ReadonlyArray` too.